### PR TITLE
docs: added documentation for hooks

### PR DESCRIPTION
**Pull request by iisakkirotko**
Added documentation for `use_effect`, `use_memo`, and improved the documentation of `use_thread`

### CODE REVIEW

**Comment by iisakkirotko** [comment]
Current dependencies on/for this PR:
* master
  * **PR #372** <a href="https://app.graphite.dev/github/pr/widgetti/solara/372?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/widgetti/solara/372?utm_source=stack-comment).

File: solara/website/pages/api/use_memo.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+```python
+def use_memo(
+    f: Any,
+    dependencies: Any | None = None,
+    debug_name: str = None
+) -> Any:
+    ...
+```
+
+`use_memo` stores ([memoize](/api/memoize)) the function return on first render, and then excludes it from being re-executed, except when one of the `dependencies` changes. `dependencies` can take the value `None`, in which case dependencies are automatically obtained from nonlocal variables. If an empty list is passed as `dependencies` instead, the function is only executed once over the entire lifetime of the component.
```
Comment (maartenbreddels) [inline solara/website/pages/api/use_memo.md:12]:
```suggestion
`use_memo` stores ([memoize](https://en.wikipedia.org/wiki/Memoization)) the function return on first render, and then excludes it from being re-executed, except when one of the `dependencies` changes. `dependencies` can take the value `None`, in which case dependencies are automatically obtained from nonlocal variables. If an empty list is passed as `dependencies` instead, the function is only executed once over the entire lifetime of the component.

Not to be confused with [memorize](https://solara.dev/api/memoize) which can cache multiple return values and which can be used outside of component.
```

**Review by maartenbreddels — CHANGES_REQUESTED** [review]
small comments, let me know if they make sense

File: solara/website/pages/api/use_memo.md (hunk 1 above)
Comment (iisakkirotko) [reply solara/website/pages/api/use_memo.md]: Not to be confused indeed.

**Comment by maartenbreddels** [comment]
Thank you 🙏

**Pull request merged**